### DISCUSSION TITLE
Do I really have to install CUDA toolkit?

### DISCUSSION THREAD

**Discussion by quasiblob** [vladmandic/sdnext#734] [Q&A] [answered]
Step two of installation says:
"If you have nVidia GPU, install nVidia CUDA toolkit:
https://developer.nvidia.com/cuda-downloads"

Do I really have to do this? 

1. I'm asking this because this is a fork of Automatic1111's web ui, and for that I didn't have to install cuda separately. I understand you may have a different installer and all that stuff.

2. If I do have to install CUDA toolkit, which version do I have to install? Link provided gives several choices for Windows (10, 11, Server 2019, Server 2022).

4. Is this CUDA toolkit a different thing than CUDA I already have installed along with NVIDIA display driver? My nvidia-smi shows that I have CUDA version 12.1 installed. 

I wouldn't want to install anything unnecessary system wide, unless a must, I like it how A1111 web ui operates mostly by installing stuff in its venv AFAIK.

**Comment by vladmandic** [comment, marked as answer]
you don't, its built-in. i need to update docs.

**Reply by quasiblob** [reply]
Thanks, 

I also watched pretty high profile AI youtuber claiming one has to install CUDA toolkit (video was 13 days old), so I started to doubt what if...

Anyway for others - Install is simple and fast:

Create a folder and open it in cmd.

Do a "git clone".

Start webui.bat when repository has been cloned.

For me, after 5 minutes the default UI was ready - and settings and configs seem to be really nicely tuned.

![image](https://user-images.githubusercontent.com/13034046/235967717-5ba44a69-fc3a-4576-8558-675b3fe0154b.png)

BTW - I've been avoiding testing your fork, as you only show dark UI images. 
But there is also the default Gradio Light theme available, which is a must for my eyes.
Might be a good idea to add an image of Light UI too? Single image would be enough.

![image](https://user-images.githubusercontent.com/13034046/235969662-058b4161-fc0a-400d-b90f-be9004b783c7.png)

**Comment by vladmandic** [comment]
default is dark, but you can pick any theme you want, its in ui settings.
